src/platform/sdl/main.c

```diff
@@ -321,3 +337,61 @@ static void mSDLDeinit(struct mSDLRenderer* renderer) {
 
 	SDL_Quit();
 }
+
+void setLogger(struct mCore* core) {
+	int fakeBool = 0;
+	bool logToStdout = true;
+	bool logToFile = false;
+
+	if (mCoreConfigGetIntValue(&core->config, "logToStdout", &fakeBool)) {
+		logToStdout = fakeBool;
+	}
+	if (mCoreConfigGetIntValue(&core->config, "logToFile", &fakeBool)) {
+		logToFile = fakeBool;
+	}
+	const char* logFile = mCoreConfigGetValue(&core->config, "logFile");
+
+	// Assign basic static variables
+	_logToStdout = logToStdout;
+	_logFile = NULL;
+	
+	if(logToFile && logFile) {
+		_logFile = VFileOpen(logFile, O_WRONLY | O_CREAT | O_APPEND);
+	}
+
+	// Create the filter
+	mLogFilterInit(&_filter);
+	mLogFilterSet(&_filter, "gba.bios", mLOG_FATAL);
```

Comment:
mLOG_FATAL is in the default filter so this isn't needed anymore.

Reply:
Hmm... Are you 100% sure? Because I tried removing it before, and I'd get a list of values that were in the registers when it went into VBlank at the end of my test ROM's execution.

If I leave that there, then I don't get that list...

Reply:
Just checked, and it makes it so you can't edit the logLevel for gba.bios by using the -l option...

Reply:
Btw, the default, if no -l is specified, is 0. So no logging whatsoever...

Reply:
That's odd, the default is supposed to be 3--fatal and error.

Reply:
Let's remove both of the log level overrides then, and manage a default log level of 3 in a later commit. It looks like it's only 3 in the Qt frontend right now.

Reply:
So, I added a default of WARN | FATAL | ERROR, which is the one QT uses for anything that isn't gba.bios and core.status, for which it uses this line and the one below this one.

The issue is that, by removing this line in particular, if you select mLOG_DEBUG (in order to see messages from the game, which is what I use the log function for), you get a log message for each single SWI call. Which can be VERY spammy.

You can see what I mean below, with before and after.

[test_prev.log](https://github.com/mgba-emu/mgba/files/7611047/test_prev.log)
[test.log](https://github.com/mgba-emu/mgba/files/7611048/test.log)

Maybe adding some sort of configuration parameter to this? Otherwise I can just keep a fork of mine and let people use that for my use case.

Reply:
That's the already existing behavior so it's not a good idea to change it in a mostly unrelated PR. That said, you can change this with e.g. -C logLevel.gba.bios=0 to turn off all BIOS logging.

Reply:
That does work in Qt, and it is... a good solution!
But not in SDL (simply because the code isn't there yet).

Will definetly look into putting it in though. Seems very useful.

Reply:
That's strange? It worked for me on master. I set -l -1 to enable all logs and then that to disable BIOS logs and it worked. Maybe the issue is local to your branch?

Reply:
Yeah, because SDL in main uses the thread.c implementation of the logger, which has this line:
mLogFilterLoad(threadContext->logger.d.filter, &core->config);

Looking at that file makes me wonder...

Should I add that line to this file, in order to keep logger and filter initialization together, or should I remove the filter initialization entirely from here, since thread.c will handle that...?

Reply:
Fixed in https://github.com/mgba-emu/mgba/pull/2345/commits/e938be4ba83a65f1012f4943af60eb195f7af610 , let me know if you'd prefer setting the filter to NULL for the logger and letting thread.c initialize it instead.